### PR TITLE
Support submetrics

### LINKED ISSUE #36: Sub Metrics support
From one query

select sum(response_time), count(*) from requests group by handler;

We should be able to get two metrics:

requests_resp_time_count
sequests_rest_time_sum

### PR DESCRIPTION
**Pull request by roidelapluie**
Fixes #36

Signed-off-by: Julien Pivotto <roidelapluie@inuits.eu>

### CODE REVIEW

**Comment by roidelapluie** [comment]
I still need to fix the way we detect if there are wultiple values per metrics so this is still WIP

**Comment by roidelapluie** [comment]
Fixed, and added tests.

**Comment by roidelapluie** [comment]
ping @haxorof @disq @bruth for review :)

File: config_test.go [hunk 1]
```diff
@@ -177,6 +177,38 @@ func Test_loadQueryConfig(t *testing.T) {
...
+		{
+			name: "queries-submetrics",
+			args: args{
+				queriesFile: "test-resources/config-test/queries-submetrics.yml",
+				config:      c,
+			},
+			want: []*Query{
+				&Query{
+					Name:          "query_ds_1",
+					SQL:           "select 1 as \"sum\", 2 as \"count\" from dual\n",
```
Comment (bruth) [inline config_test.go]: Prefer literal strings (backticks) when using double quotes in a string.

```go
`select 1 as "sum" ...`
```

File: set.go [hunk 2]
```diff
@@ -75,35 +80,58 @@ func (r *QueryResult) SetMetrics(recs records) (map[string]bool, error) {
...
+			for k, v := range row {
+				if len(row) > 1 && strings.ToLower(k) != datafield { // facet field, add to facets
+					submetric := false
+					for _, n := range submetrics {
+						if strings.ToLower(k) == n {
+							submetric = true
+						}
+					}
+					// it is a facet field and not a submetric field
+					if submetric == false {
```
Comment (bruth) [inline set.go]: Prefer `if !submetric {` rather than equality to `false`.

**Comment by bruth** [comment]
Looks good otherwise!

**Comment by roidelapluie** [comment]
@bruth I addressed your comments.

**Comment by bruth** [comment]
👍 Nice work

**Pull request merged**